system-test/src/test/java/org/apache/mesos/logstash/systemtest/DeploymentSystemTest.java

```diff
@@ -83,121 +92,115 @@ protected State getClusterStateInfo() {
-        await().atMost(1, TimeUnit.MINUTES).pollDelay(1, TimeUnit.SECONDS).ignoreExceptions().until(() -> {
-            assertTrue(dockerClient.inspectContainerCmd(logstashSlave).exec().getState().isRunning());
-
-            final CreateContainerResponse loggerContainer = dockerClient.createContainerCmd("ubuntu:15.10").withLinks(new Link(logstashSlave, "logstash")).withCmd("logger", "--server=logstash", "--port=" + sysLogPort, "--udp", "--rfc3164", randomLogLine).exec();
-            dockerClient.startContainerCmd(loggerContainer.getId()).exec();
-            await().atMost(10, TimeUnit.SECONDS).until(() -> {
-                // TODO: this is a hack to determine whether the container has stopped.
-                // We should use ...exec().getState().getRunning() but docker-java doesn't provide that
-                // (even though it's available in the JSON provided by Docker).
-                final String finishedAt = dockerClient.inspectContainerCmd(loggerContainer.getId()).exec().getState().getFinishedAt();
-                return StringUtils.isNotBlank(finishedAt) && !finishedAt.equals("0001-01-01T00:00:00Z");
-            });
-            final int exitCode = dockerClient.inspectContainerCmd(loggerContainer.getId()).exec().getState().getExitCode();
-            dockerClient.removeContainerCmd(loggerContainer.getId()).exec();
-            assertEquals(0, exitCode);
-            elasticsearchClient.get().prepareSearch("logstash-*").setQuery(QueryBuilders.simpleQueryStringQuery("hello")).addField("message").addField("mesos_slave_id").execute().actionGet().getHits().getAt(0).fields();
-        });
-        await().atMost(1, TimeUnit.MINUTES).pollDelay(1, TimeUnit.SECONDS).until(() -> {
-            Map<String, SearchHitField> fields = elasticsearchClient.get().prepareSearch("logstash-*").setQuery(QueryBuilders.simpleQueryStringQuery("hello")).addField("message").addField("mesos_slave_id").execute().actionGet().getHits().getAt(0).fields();
-
-            String esMessage = fields.get("message").getValue();
-            assertEquals(randomLogLine, esMessage.trim());
+        String logline = "Hello " + RandomStringUtils.randomAlphanumeric(32);
 
-            String esMesosSlaveId = fields.get("mesos_slave_id").getValue();
+        Thread thread = new Thread() {
+            @Override
+            public void run() {
+                try {
+                    DatagramSocket socket = null;
+                    socket = new DatagramSocket();
+                    byte[] buf = logline.getBytes();
+                    InetAddress address = InetAddress.getByName("localhost");
+                    DatagramPacket packet = new DatagramPacket(buf, buf.length, address, 514);
+                    socket.send(packet);
+                } catch (IOException e) {
+                    // Ignore
+                }
+            }
+        };
+        thread.run();
```

Comment:
Not sure I understand the purpose of this bit? -Oh and BTW, running `Thread.run()` rather than `Thread.start()` will actually inline it on the same thread.